### PR TITLE
feat: POM-202: [f] - usunięcie ogłoszenia z poziomu konta - popup

### PR DESCRIPTION
**Pull request by pswiderski123**

### CODE REVIEW

File: src/app/my-account/my-account/my-account.component.ts [hunk 1]
```diff
@@ -9,23 +9,26 @@ import {
   UserInfo,
   UsersResourceService,
 } from '@app/core/api';
-import { Observable, switchMap, tap } from 'rxjs';
+import { switchMap, take, tap } from 'rxjs';
 import { Router } from '@angular/router';
 import { CategoryRoutingName, CorePath } from '@app/shared/models';
+import { MatDialog } from '@angular/material/dialog';
+import { ConfirmRemoveAdComponent } from '@app/shared/components/dialogs/confirm-remove-ad/confirm-remove-ad.component';
```
Comment (bbartosiak) [inline src/app/my-account/my-account/my-account.component.ts:16]: @app/shared/components

File: src/app/shared/shared.module.ts [hunk 2]
```diff
@@ -7,10 +7,18 @@ import { RouterModule } from '@angular/router';
 import { MatIconModule } from '@angular/material/icon';
 import { TranslateModule } from '@ngx-translate/core';
 import { ValidatorsDirectivesModule } from './validators';
+import { ConfirmRemoveAdComponent } from '@app/shared/components/dialogs/confirm-remove-ad/confirm-remove-ad.component';
```
Comment (bbartosiak) [inline src/app/shared/shared.module.ts:10]: @app/shared/components

File: src/app/shared/components/dialogs/confirm-remove-ad/confirm-remove-ad.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { ConfirmRemoveAdComponent } from '@app/shared/components/dialogs/confirm-remove-ad/confirm-remove-ad.component';
```
Comment (bbartosiak) [inline src/app/shared/components/dialogs/confirm-remove-ad/confirm-remove-ad.component.spec.ts:3]: @app/shared/components

File: src/app/shared/components/dialogs/confirm-remove-ad/confirm-remove-ad.component.scss [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      letter-spacing: 1.25px;
+      color: #052c65;
+      text-transform: uppercase;
+    }
+  }
+
+  button:first-of-type {
+    span {
+      color: #ffffff;
+    }
```
Comment (neu5) [inline src/app/shared/components/dialogs/confirm-remove-ad/confirm-remove-ad.component.scss:43]: Może da się skorzystać z globalnych styli?
https://github.com/coi-gov-pl/gui-web-pomogamukrainie-forum/blob/main/src/assets/styles/_vars.scss

**Review by bbartosiak — APPROVED** [review]

File: src/app/my-account/my-account/my-account.component.ts [hunk 5]
```diff
@@ -43,7 +46,21 @@ export class MyAccountComponent implements OnInit {
...
+      maxHeight: '450px',
+      maxWidth: '720px',
+    });
+
+    dialogRef.componentInstance.currentAnnouncement = announcement;
+
+    dialogRef.componentInstance.onClosed.pipe(take(1)).subscribe((confirmed: boolean) => {
+      dialogRef.close();
+      if (confirmed) {
+        // TODO: call to api
```
Comment (mgol) [inline src/app/my-account/my-account/my-account.component.ts:61]: Is API not ready yet?

File: src/app/my-account/confirm-remove-ad/confirm-remove-ad.component.html [hunk 6]
```diff
@@ -0,0 +1,52 @@
+<div class="container">
+  <div class="icon">
+    <mat-icon fontSet="material-icons-outlined">restore_from_trash</mat-icon>
+  </div>
+  <div class="title">{{ "CONFIRM_REMOVE_AD" | translate }}</div>
+
+  <app-search-result
+    [location]="$any(currentAnnouncement).location"
```
Comment (mgol) [inline src/app/my-account/confirm-remove-ad/confirm-remove-ad.component.html:8]: Why so many `$any`s? That defeats the point of type checking.

File: src/app/my-account/my-account/my-account.component.ts (hunk 5 above)
Comment (pswiderski123) [reply src/app/my-account/my-account/my-account.component.ts:61]: polaczenie z API bedzie developowane w oddzielnym zadaniu

**Review by mgol — APPROVED** [review]
LGTM. Usage of `$any` is unfortunate but it's a result of backend not sending the offer type. I hope they'll fix it soon and we'll be able to drop the `$any`s then.

**Pull request merged**